azurerm/resource_arm_dev_test_lab_schedule.go

```diff
@@ -0,0 +1,489 @@
+	d.Set("resource_group_name", resGroup)
+
+	if props := resp.ScheduleProperties; props != nil {
+		if timeZoneId := props.TimeZoneID; *timeZoneId != "" {
+			d.Set("time_zone_id", *timeZoneId)
+		}
+
+		switch status := props.Status; status {
+		case dtl.EnableStatusEnabled:
+			d.Set("status", string(dtl.EnableStatusEnabled))
+		case dtl.EnableStatusDisabled:
+			d.Set("status", string(dtl.EnableStatusDisabled))
+		default:
+		}
+
+		if weeklyRecurrence := props.WeeklyRecurrence; weeklyRecurrence != nil {
+			if err := d.Set("weekly_recurrence", flattenAzureRmDevTestLabScheduleRecurrenceWeekly(props.WeeklyRecurrence)); err != nil {
+				return fmt.Errorf("Error setting `weeklyRecurrence`: %#v", err)
+			}
+		}
+
+		if dailyRecurrence := props.DailyRecurrence; dailyRecurrence != nil {
+			if err := d.Set("daily_recurrence", flattenAzureRmDevTestLabScheduleRecurrenceDaily(props.DailyRecurrence)); err != nil {
+				return fmt.Errorf("Error setting `dailyRecurrence`: %#v", err)
+			}
+		}
+
+		if hourlyRecurrence := props.HourlyRecurrence; hourlyRecurrence != nil {
+			if err := d.Set("hourly_recurrence", flattenAzureRmDevTestLabScheduleRecurrenceHourly(props.HourlyRecurrence)); err != nil {
+				return fmt.Errorf("Error setting `dailyRecurrence`: %#v", err)
+			}
+		}
+
+		if err := d.Set("notification_settings", flattenAzureRmDevTestLabScheduleNotificationSettings(props.NotificationSettings)); err != nil {
+			return fmt.Errorf("Error setting `notificationSettings`: %#v", err)
+		}
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmDevTestLabSchedulesDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).vmExtensionClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["schedules"]
+	devTestLabName := id.Path["labs"]
+
+	future, err := client.Delete(ctx, resGroup, devTestLabName, name)
+	if err != nil {
+		return err
+	}
+
+	return future.WaitForCompletionRef(ctx, client.Client)
+}
+
+func expandArmDevTestLabScheduleRecurrenceDaily(d *schema.ResourceData) (*dtl.DayDetails, error) {
+	dailyRecurrenceConfigs := d.Get("daily_recurrence").(*schema.Set).List()
+	dailyRecurrenceConfig := dailyRecurrenceConfigs[0].(map[string]interface{})
+	dailyTime := dailyRecurrenceConfig["time"].(string)
+
+	return &dtl.DayDetails{
+		Time: &dailyTime,
+	}, nil
+
+}
+
+func flattenAzureRmDevTestLabScheduleRecurrenceDaily(dailyRecurrence *dtl.DayDetails) []interface{} {
+	if dailyRecurrence == nil {
+		return []interface{}{}
+	}
+
+	result := make(map[string]interface{})
+
+	if dailyRecurrence.Time != nil {
+		result["time"] = *dailyRecurrence.Time
+	}
+
+	return []interface{}{result}
+}
+
+func expandArmDevTestLabScheduleRecurrenceWeekly(d *schema.ResourceData) (*dtl.WeekDetails, error) {
+	weeklyRecurrenceConfigs := d.Get("weekly_recurrence").(*schema.Set).List()
```

Comment:
(same here) as this is a list we can make this:

```suggestion
	weeklyRecurrenceConfigs := d.Get("weekly_recurrence").([]interface{})
```

Reply:
Done